File: software/examples/go/example_button_callback.go
package main

import (
	"fmt"
	"github.com/Tinkerforge/go-api-bindings/ipconnection"
	"github.com/Tinkerforge/go-api-bindings/lcd_20x4_bricklet"
)

const ADDR string = "localhost:4223"
const UID string = "XYZ" // Change XYZ to the UID of your LCD 20x4 Bricklet.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	lcd, _ := lcd_20x4_bricklet.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	lcd.RegisterButtonPressedCallback(func(button uint8) {
		fmt.Printf("Button Pressed: %d\n", button)
	})

	lcd.RegisterButtonReleasedCallback(func(button uint8) {
		fmt.Printf("Button Released: %d\n", button)
	})

	fmt.Print("Press enter to exit.")
	fmt.Scanln()
}

File: software/examples/rust/example_button_callback.rs
use std::{error::Error, io, thread};
use tinkerforge::{ipconnection::IpConnection, lcd_20x4_bricklet::*};

const HOST: &str = "127.0.0.1";
const PORT: u16 = 4223;
const UID: &str = "XYZ"; // Change XYZ to the UID of your LCD 20x4 Bricklet

fn main() -> Result<(), Box<dyn Error>> {
    let ipcon = IpConnection::new(); // Create IP connection
    let lcd_20x4_bricklet = LCD20x4Bricklet::new(UID, &ipcon); // Create device object

    ipcon.connect(HOST, PORT).recv()??; // Connect to brickd
                                        // Don't use device before ipcon is connected

    //Create listener for button pressed events.
    let button_pressed_listener = lcd_20x4_bricklet.get_button_pressed_receiver();
    // Spawn thread to handle received events. This thread ends when the lcd_20x4_bricklet
    // is dropped, so there is no need for manual cleanup.
    thread::spawn(move || {
        for event in button_pressed_listener {
            println!("Button Pressed: {}", event);
        }
    });

    //Create listener for button released events.
    let button_released_listener = lcd_20x4_bricklet.get_button_released_receiver();
    // Spawn thread to handle received events. This thread ends when the lcd_20x4_bricklet
    // is dropped, so there is no need for manual cleanup.
    thread::spawn(move || {
        for event in button_released_listener {
            println!("Button Released: {}", event);
        }
    });

    println!("Press enter to exit.");
    let mut _input = String::new();
    io::stdin().read_line(&mut _input)?;
    ipcon.disconnect();
    Ok(())
}
